pkg/types/builder.go

```diff
@@ -89,11 +96,19 @@ func (g *Builder) buildResource(res *schema.Resource, names ...string) (*types.N
 		if comment.TerrajetOptions.FieldJSONTag != nil {
 			jsonTag = *comment.TerrajetOptions.FieldJSONTag
 		}
-
-		fieldType, err := g.buildSchema(sch, append(names, fieldName.Camel))
+		fieldType, err := g.buildSchema(sch, refs, append(names, fieldName.Camel))
 		if err != nil {
 			return nil, nil, errors.Wrapf(err, "cannot infer type from schema of field %s", fieldName.Snake)
 		}
+
+		// Build the field path based on the fact that names is a slice of
+		// strings which contains the name of the parent fields sequentially.
+		// We skip the first element since it is the main CRD type.
+		fp := strings.Join(append(names[1:], fieldName.Camel), ".")
```

Comment:
We discussed online about whether having users give the field path or path in the `<type>.<field>` format which is what we need at minimum. It seems like it would work in both cases. For consistency of inputs, we're going with this implementation.